crates/ironrdp-cliprdr-native/src/windows/utils.rs

```diff
@@ -0,0 +1,67 @@
+use ironrdp_cliprdr::pdu::ClipboardFormatId;
+use tracing::error;
+use windows::Win32::Foundation::{GetLastError, HANDLE, HGLOBAL, WIN32_ERROR};
+use windows::Win32::System::DataExchange::SetClipboardData;
+use windows::Win32::System::Memory::{GlobalAlloc, GlobalFree, GlobalLock, GlobalUnlock, GMEM_MOVEABLE};
+
+use crate::windows::WinCliprdrResult;
+
+/// Safe wrapper around windows global memory buffer.
+struct GlobalMemoryBuffer(HGLOBAL);
+
+impl GlobalMemoryBuffer {
+    /// Creates new global memory buffer and copies data into it.
+    fn from_slice(data: &[u8]) -> WinCliprdrResult<Self> {
+        // SAFETY: GlobalAlloc will return null only if there is not enough memory to allocate
+        // `windows` crate will catch this error via internal invalid handle check
+        let handle = unsafe { GlobalAlloc(GMEM_MOVEABLE, data.len())? };
+
+        // SAFETY: We own the handle, so it is safe to assume that GlobalLock will succeed
+        unsafe {
+            let dst = GlobalLock(handle);
+            std::ptr::copy(data.as_ptr(), dst as _, data.len());
+            GlobalUnlock(handle);
+        };
```

Comment:
More elaborated safety comments + use `copy_nonoverlapping`
```suggestion
        // SAFETY: We created the handle and ensured it wasn’t null just above.
        // Note that we don’t check for failure because we own the handle and
        // know that the specified memory block can’t be discarded at this point.
        let dst = unsafe { GlobalLock(handle) };
        // SAFETY:
        // - `data` is valid for reads of `data.len()` bytes.
        // - `dst` is valid for writes of `data.len()` bytes, we allocated enough above.
        // - Both `data` and `dst` are properly aligned: u8 alignment is 1
        // - Memory regions are not overlapping, `dst` was allocated by us just above.
        unsafe { std::ptr::copy_nonoverlapping(data.as_ptr(), dst as _, data.len()) };
        // SAFETY: We called `GlobalLock` on this handle just above.
        unsafe { GlobalUnlock(handle) };
```